### ISSUE TITLE
DOMParser should return XMLDocument if applicable

### ISSUE THREAD

**Issue by jp7837** [jsdom/jsdom#2971]
### Basic info:

- **Node.js version:** 12.8.1
- **jsdom version:** 16.2.2

### Minimal reproduction case

```js
const parser = new DOMParser();
const doc = parser.parseFromString(myXml, "text/xml");

// This would fail in Jest since it's a Document
expect(doc).toBeInstanceOf(XMLDocument);
```

### How does similar code behave in browsers?

Returns `XMLDocument`

**Comment by jp7837** [comment]
Looks like this could just use the fix from #2841 (cc @domenic)

**Comment by TimothyGu** [comment]
Indeed, this appears to be fixed in jsdom. If you are using Jest then you should search how to use current version of jsdom with Jest.

**Comment by jp7837** [comment]
@TimothyGu this is with the latest version of jsdom (16.2.2), Jest is already consuming it.  Can we reopen?

**Comment by TimothyGu** [comment]
Ah okay, I misunderstood https://github.com/jsdom/jsdom/issues/2971#issuecomment-627429112, sorry about that.

**Comment by domenic** [comment]
jsdom is currently behaving as per the spec https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#domparser, so this will need to wait on the corresponding spec changes before we'd update jsdom.

**Comment by TWilson023** [comment]
The spec does indicate `parseFromString` should return a `Document`, but `Document` is an interface.  Would it be against spec to return a more specific extension (in this case, `XMLDocument`), as all modern browsers seem to do?

**Comment by domenic** [comment]
"interface" just means "class" in spec-speak. So yeah, it'd be against the spec to return the XMLDocument class/interface.
